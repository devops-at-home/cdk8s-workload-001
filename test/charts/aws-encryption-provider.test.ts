import { App } from 'cdk8s';
import {
    AwsEncryptionProvider,
    parseInputs,
    AwsEncryptionProviderProps,
} from '../../src/charts/aws-encryption-provider.chart';

describe('Snapshot tests', () => {
    test('Hello Kubernetes', () => {
        const app = new App();

        const chart = new AwsEncryptionProvider(app, 'AwsEncryptionProvider');

        expect(chart).toMatchSnapshot();
    });

    test('parseInputs function default values', () => {
        const response = parseInputs({});
        expect(response.containerPort).toStrictEqual(8080);
        expect(response.image).toStrictEqual(
            '012345678901.dkr.ecr.ap-southeast-2.amazonaws.com/aws-encryption-provider:latest'
        );
        expect(response.name).toStrictEqual('aws-encryption-provider');
        expect(response.region).toStrictEqual('ap-southeast-2');
        expect(response.key).toStrictEqual(
            'arn:aws:kms:ap-southeast-2:012345678901:key/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
        );
    });

    test('parseInputs function with inputs', () => {
        const inputs: AwsEncryptionProviderProps = {
            containerPort: 8081,
            image: 'some-image',
            name: 'some-name',
            account: '123456789012',
            key: 'xx',
            ecr: false,
            region: 'some-region',
        };

        const response = parseInputs(inputs);
        expect(response.containerPort).toStrictEqual(inputs.containerPort);
        expect(response.image).toStrictEqual(inputs.image);
        expect(response.name).toStrictEqual(inputs.name);
        expect(response.key).toStrictEqual(
            `arn:aws:kms:${inputs.region}:${inputs.account}:key/${inputs.key}`
        );
        expect(response.region).toStrictEqual(inputs.region);
    });
});
