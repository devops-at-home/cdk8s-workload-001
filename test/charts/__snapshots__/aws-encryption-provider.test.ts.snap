// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Hello Kubernetes 1`] = `
AwsEncryptionProvider {
  "_disableResourceNameHashes": false,
  "_labels": {},
  "namespace": undefined,
  "node": Node {
    "_children": {
      "deployment": KubeDeployment {
        "apiGroup": "apps",
        "apiVersion": "apps/v1",
        "chart": [Circular],
        "kind": "Deployment",
        "metadata": ApiObjectMetadataDefinition {
          "_additionalAttributes": {
            "labels": {
              "app.kubernetes.io/name": "aws-encryption-provider",
            },
            "name": "aws-encryption-provider",
            "namespace": "kube-system",
          },
          "annotations": {},
          "finalizers": [],
          "labels": {
            "app.kubernetes.io/name": "aws-encryption-provider",
          },
          "name": "aws-encryption-provider",
          "namespace": "kube-system",
          "ownerReferences": [],
        },
        "name": "aws-encryption-provider",
        "node": Node {
          "_children": {},
          "_context": {},
          "_dependencies": Set {},
          "_locked": false,
          "_metadata": [],
          "_validations": [],
          "host": [Circular],
          "id": "deployment",
          "scope": [Circular],
        },
        "patches": [],
        "props": {
          "apiVersion": "apps/v1",
          "kind": "Deployment",
          "metadata": {
            "labels": {
              "app.kubernetes.io/name": "aws-encryption-provider",
            },
            "name": "aws-encryption-provider",
            "namespace": "kube-system",
          },
          "spec": {
            "replicas": 1,
            "selector": {
              "matchLabels": {
                "app.kubernetes.io/name": "aws-encryption-provider",
              },
            },
            "template": {
              "metadata": {
                "labels": {
                  "app.kubernetes.io/name": "aws-encryption-provider",
                },
                "name": "aws-encryption-provider",
                "namespace": "kube-system",
              },
              "spec": {
                "containers": [
                  {
                    "command": [
                      "/aws-encryption-provider",
                      "--key=arn:aws:kms:ap-southeast-2:012345678901:key/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
                      "--region=ap-southeast-2",
                      "--listen=/var/run/kmsplugin/socket.sock",
                    ],
                    "image": "012345678901.dkr.ecr.ap-southeast-2.amazonaws.com/aws-encryption-provider:latest",
                    "imagePullPolicy": "IfNotPresent",
                    "livenessProbe": {
                      "httpGet": {
                        "path": "/healthz",
                        "port": IntOrString {
                          "value": 8080,
                        },
                      },
                    },
                    "name": "aws-encryption-provider",
                    "ports": [
                      {
                        "containerPort": 8080,
                        "name": "http",
                        "protocol": "TCP",
                      },
                    ],
                    "volumeMounts": [
                      {
                        "mountPath": "/var/run/kmsplugin",
                        "name": "var-run-kmsplugin",
                      },
                      {
                        "mountPath": "/root/.aws",
                        "name": "awsssm",
                      },
                    ],
                  },
                ],
                "imagePullSecrets": [
                  {
                    "name": "ecrauth",
                  },
                ],
                "serviceAccountName": "aws-encryption-provider",
                "volumes": [
                  {
                    "hostPath": {
                      "path": "/var/run/kmsplugin",
                      "type": "DirectoryOrCreate",
                    },
                    "name": "var-run-kmsplugin",
                  },
                  {
                    "hostPath": {
                      "path": "/root/.aws",
                    },
                    "name": "awsssm",
                  },
                ],
              },
            },
          },
        },
        Symbol(@aws-cdk/core.DependableTrait): {
          "dependencyRoots": [
            [Circular],
          ],
        },
      },
      "serviceAccount": KubeServiceAccount {
        "apiGroup": "core",
        "apiVersion": "v1",
        "chart": [Circular],
        "kind": "ServiceAccount",
        "metadata": ApiObjectMetadataDefinition {
          "_additionalAttributes": {
            "labels": {
              "app.kubernetes.io/name": "aws-encryption-provider",
            },
            "name": "aws-encryption-provider",
            "namespace": "kube-system",
          },
          "annotations": {},
          "finalizers": [],
          "labels": {
            "app.kubernetes.io/name": "aws-encryption-provider",
          },
          "name": "aws-encryption-provider",
          "namespace": "kube-system",
          "ownerReferences": [],
        },
        "name": "aws-encryption-provider",
        "node": Node {
          "_children": {},
          "_context": {},
          "_dependencies": Set {},
          "_locked": false,
          "_metadata": [],
          "_validations": [],
          "host": [Circular],
          "id": "serviceAccount",
          "scope": [Circular],
        },
        "patches": [],
        "props": {
          "apiVersion": "v1",
          "kind": "ServiceAccount",
          "metadata": {
            "labels": {
              "app.kubernetes.io/name": "aws-encryption-provider",
            },
            "name": "aws-encryption-provider",
            "namespace": "kube-system",
          },
        },
        Symbol(@aws-cdk/core.DependableTrait): {
          "dependencyRoots": [
            [Circular],
          ],
        },
      },
    },
    "_context": {},
    "_dependencies": Set {},
    "_locked": false,
    "_metadata": [],
    "_validations": [],
    "host": [Circular],
    "id": "AwsEncryptionProvider",
    "scope": App {
      "node": Node {
        "_children": {
          "AwsEncryptionProvider": [Circular],
        },
        "_context": {},
        "_dependencies": Set {},
        "_locked": false,
        "_metadata": [],
        "_validations": [],
        "host": [Circular],
        "id": "",
        "scope": undefined,
      },
      "outdir": "dist",
      "outputFileExtension": ".k8s.yaml",
      "recordConstructMetadata": false,
      "yamlOutputType": 1,
      Symbol(@aws-cdk/core.DependableTrait): {
        "dependencyRoots": [
          [Circular],
        ],
      },
    },
  },
  Symbol(@aws-cdk/core.DependableTrait): {
    "dependencyRoots": [
      [Circular],
    ],
  },
}
`;
